clustering/kubernetes.py

```diff
@@ -0,0 +1,436 @@
+# 'create a'. The script then iterates over all of these base objects
+# to get the endpoint URL and was used to generate the KIND_URL map.
+#
+# import json
+# from urllib2 import urlopen
+#
+# r = urlopen("https://raw.githubusercontent.com/kubernetes"
+#            "/kubernetes/master/api/swagger-spec/v1.json")
+# v1 = json.load(r)
+#
+# apis = {}
+# for a in v1['apis']:
+#     p = a['path']
+#     for o in a['operations']:
+#         if o["summary"].startswith("create a") and o["type"] != "v1.Event":
+#             apis[o["type"]] = p
+#
+# def print_kind_url_map():
+#     results = []
+#     for a in apis.keys():
+#         results.append('"%s": "%s"' % (a[3:].lower(), apis[a]))
+#     results.sort()
+#     print "KIND_URL = {"
+#     print ",\n".join(results)
+#     print "}"
+#
+# if __name__ == '__main__':
+#     print_kind_url_map()
+############################################################################
+############################################################################
+
+KIND_URL = {
+    "binding": "/api/v1/namespaces/{namespace}/bindings",
+    "endpoints": "/api/v1/namespaces/{namespace}/endpoints",
+    "limitrange": "/api/v1/namespaces/{namespace}/limitranges",
+    "namespace": "/api/v1/namespaces",
+    "node": "/api/v1/nodes",
+    "persistentvolume": "/api/v1/persistentvolumes",
+    "persistentvolumeclaim":
+        "/api/v1/namespaces/{namespace}/persistentvolumeclaims",
+    "pod": "/api/v1/namespaces/{namespace}/pods",
+    "podtemplate": "/api/v1/namespaces/{namespace}/podtemplates",
+    "replicationcontroller":
+        "/api/v1/namespaces/{namespace}/replicationcontrollers",
+    "resourcequota": "/api/v1/namespaces/{namespace}/resourcequotas",
+    "secret": "/api/v1/namespaces/{namespace}/secrets",
+    "service": "/api/v1/namespaces/{namespace}/services",
+    "serviceaccount": "/api/v1/namespaces/{namespace}/serviceaccounts"
+}
+USER_AGENT = "ansible-k8s-module/0.0.1"
+
+
+# TODO(erjohnso): SSL Certificate validation is currently unsupported.
+# It can be made to work when the following are true:
+# - Ansible consistently uses a "match_hostname" that supports IP Address
+#   matching. This is now true in >= python3.5.0. Currently, this feature
+#   is not yet available in backports.ssl_match_hostname (still 3.4).
+# - Ansible allows passing in the self-signed CA cert that is created with
+#   a kubernetes master. The lib/ansible/module_utils/urls.py method,
+#   SSLValidationHandler.get_ca_certs() needs a way for the Kubernetes
+#   CA cert to be passed in and included in the generated bundle file.
+# When this is fixed, the following changes can be made to this module,
+# - Remove the 'return' statement in line 254 below
+# - Set 'required=true' for certificate_authority_data and ensure that
+#   ansible's SSLValidationHandler.get_ca_certs() can pick up this CA cert
+# - Set 'required=true' for the validate_certs param.
+def _decode_cert_data(module):
```

Comment:
Does it make sense to mark this function as 'private' (underscore in front of the function name)? Even all the functions would be wrapped into a class, I would not make it 'private' either.